# Fixes a crash with fill_read_buffer(...) Buffer overflow" if peer sends
# both encryption pads and their total size does not allow the subsequent
# bitfield/protocol message to be received correctly. Ticket #1337.
Index: libtorrent/src/protocol/handshake.h
===================================================================
--- libtorrent/src/protocol/handshake.h	(revision 1073)
+++ libtorrent/src/protocol/handshake.h	(working copy)
@@ -66,7 +66,7 @@
   static const uint32_t enc_pad_size         = 512;
   static const uint32_t enc_pad_read_size    = 96 + enc_pad_size + 20;
 
-  static const uint32_t buffer_size = enc_pad_read_size + 20 + enc_negotiation_size + enc_pad_size + 2 + handshake_size;
+  static const uint32_t buffer_size = enc_pad_read_size + 20 + enc_negotiation_size + enc_pad_size + 2 + handshake_size + 5;
 
   typedef ProtocolBuffer<buffer_size> Buffer;
 
